src/pkg/services/m365/api/teams.go

```diff
@@ -0,0 +1,161 @@
+package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/internal/common/tform"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/fault"
+	msgraphgocore "github.com/microsoftgraph/msgraph-sdk-go-core"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+)
+
+const teamService = "Team"
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Groups() Groups {
+	return Groups{c}
+}
+
+// On creation of each Microsoft Teams a corrsponding group gets created from them.
+// Most of the information like events, drive and mail info will be fetched directly
+// from groups. So we pull in group and process only the once which are associated with
+// a team for further proccessing of teams.
+
+// Teams is an interface-compliant provider of the client.
+type Groups struct {
+	Client
+}
+
+// GetAll retrieves all groups.
+func (c Groups) GetAll(
+	ctx context.Context,
+	filterTeams bool,
+	errs *fault.Bus,
+) ([]models.Groupable, error) {
+	service, err := c.Service()
+	if err != nil {
+		return nil, err
+	}
+
+	return getGroups(ctx, filterTeams, errs, service)
+}
+
+// GetAll retrieves all groups.
+func getGroups(
+	ctx context.Context,
+	filterTeams bool,
```

Comment:
Exposed two funcs now- 
- func (c Groups) GetAllTeams() - gets all the groups which have a corresponding team with them.
- func (c Groups) GetAll() - gets all the groups in org

Reply:
See my other comment: I think we'll want two clients.  One for teams, and one for groups, and the teams client can call groups api under the hood.  For this PR, since we don't have a need for generalized groups yet, can you just shift the api client naming to teams?  Thanks for handling all the-back-and-forth.

Reply:
As of now I have added the Teams client with uses group api under the hood.